core/src/main/java/com/radixdlt/consensus/vertexstore/VertexStoreJavaImpl.java

```diff
@@ -302,91 +339,102 @@ public InsertVertexChainResult insertVertexChain(VertexChain vertexChain) {
         }
       }
 
-      insertVertex(v).onPresent(bftInsertUpdates::add);
+      final var insertRes = insertVertexInternal(v);
+      if (insertRes.isSuccess()) {
+        bftInsertUpdates.add(insertRes.unwrap());
+      } else {
+        switch (insertRes.unwrapError()) {
+          case ALREADY_PRESENT, PREPARE_FAILED -> {
+            // No-op, continue iterating the vertices
+          }
+          case VERTEX_STORE_SIZE_EXCEEDED -> {
+            // Stop if we hit the size limit
+            break vertices_loop;
```

Comment:
Yeah this isn't ideal but it's implicitly handled in bft sync, it just goes on with the sync process (that is: `syncToQC`) regardless of the result. I'll add a comment so that we don't forget to handle that better in the next bft sync implementation :smile:

Reply:
But what would happen if this arises? Would sync essentially fail / get stuck in a weird state?
Might it be better to throw and crash the node than get stuck in a weird half-live state?